src/view/adminhtml/web/js/datepicker.js

```diff
@@ -0,0 +1,22 @@
+//This script fixes incorrect positioning in modal datepickers.
+//This file should be removed after magento fixes the datepickers.
+
+require(["jquery"], function ($) {
+    $(window).on("load", function () {
+        if (window.location.pathname.includes("/admin/catalog/product/edit/id")) {
+            (($) => {
+                const show = $.datepicker._showDatepicker;
+                const datePicker = $($.datepicker.dpDiv[0]);
+                $.extend($.datepicker, {
+                    _showDatepicker: (trigger) => {
+                        show(trigger);
+
+                        if ($(trigger).parents(".modal-slide").length > 0) {
+                            datePicker.css("top", `${datePicker.height() - 25}px`);
```

Comment:
These calculations look like magic and aren't related to the actual trigger position.
I believe, this will fail if the datepicker input will change its position on the screen.
Please calculate the top position of the datepicker based on the position of the trigger element.

Reply:
The problem is that the original implementation also uses the triggers offset and strangely it's different for every scroll position, and our QA said that this datepicker issue occurs only here, other date pickers are not in the modals.

Reply:
If some plugin or update will add new fields in this modal, your magic calculation will fail.
Basically, you have fixed values, since the height of the datepicker doesn't change.
Please find the different approach to determine the correct position for the datepicker.

Reply:
How about:
```javascript
const { top, height } = trigger.getBoundingClientRect();
datePicker.css('top', top + height);
```